kotlin-eclipse-core/src/org/jetbrains/kotlin/core/utils/ProjectUtils.kt

```diff
@@ -0,0 +1,331 @@
+/*******************************************************************************
+ * Copyright 2000-2014 JetBrains s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.jetbrains.kotlin.core.utils
+
+import org.eclipse.core.resources.*
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.core.runtime.FileLocator
+import org.eclipse.core.runtime.IPath
+import org.eclipse.core.runtime.Platform
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.jdt.core.IJavaProject
+import org.eclipse.jdt.core.JavaCore
+import org.eclipse.jdt.core.JavaModelException
+import org.jetbrains.kotlin.core.KotlinClasspathContainer
+import org.jetbrains.kotlin.core.builder.KotlinPsiManager
+import org.jetbrains.kotlin.core.log.KotlinLogger
+import org.jetbrains.kotlin.core.model.KotlinNature
+import org.jetbrains.kotlin.psi.KtFile
+import java.io.File
+import java.io.IOException
+import java.util.*
+
+object ProjectUtils {
+
+    private const val LIB_FOLDER = "lib"
+    private const val LIB_EXTENSION = "jar"
+
+    private const val MAVEN_NATURE_ID = "org.eclipse.m2e.core.maven2Nature"
+    private const val GRADLE_NATURE_ID = "org.eclipse.buildship.core.gradleprojectnature"
+
+    @JvmStatic
+    val KT_HOME = ktHome
+
+    val accessibleKotlinProjects: List<IProject>
+        get() = ResourcesPlugin.getWorkspace().root.projects.filter { project -> isAccessibleKotlinProject(project) }
+
+    private val ktHome: String
+        get() = try {
+            val compilerBundle = Platform.getBundle("org.jetbrains.kotlin.bundled-compiler")
+            FileLocator.toFileURL(compilerBundle.getEntry("/")).file
+        } catch (e: IOException) {
+            KotlinLogger.logAndThrow(e)
+        }
+
+    fun getJavaProjectFromCollection(files: Collection<IFile>): IJavaProject? {
+        var javaProject: IJavaProject? = null
+        for (file in files) {
+            javaProject = JavaCore.create(file.project)
+            break
+        }
+
+        return javaProject
```

Comment:
Let's rewrite it to something like `files.firstOrNull()?.let { create(it.project) }`